source/postprocess/crystal_preferred_orientation.cc

```diff
@@ -0,0 +1,833 @@
+                                               this->get_mpi_communicator(),
+                                               true);
+
+          write_in_background_thread = prm.get_bool("Write in background thread");
+          temporary_output_location = prm.get("Temporary output location");
+
+          if (temporary_output_location != "")
+            {
+              // Check if a command-processor is available by calling system() with a
+              // null pointer. System is guaranteed to return non-zero if it finds
+              // a terminal and zero if there is none (like on the compute nodes of
+              // some cluster architectures, e.g. IBM BlueGene/Q)
+              AssertThrow(system((char *)nullptr) != 0,
+                          ExcMessage("Usage of a temporary storage location is only supported if "
+                                     "there is a terminal available to move the files to their final location "
+                                     "after writing. The system() command did not succeed in finding such a terminal."));
+            }
+
+          std::vector<std::string> write_raw_cpo_list = Utilities::split_string_list(prm.get("Write out raw cpo data"));
+          write_raw_cpo.resize(write_raw_cpo_list.size());
+          bool found_euler_angles = false;
+          for (unsigned int i = 0; i < write_raw_cpo_list.size(); ++i)
+            {
+              std::vector<std::string> split_raw_cpo_instructions = Utilities::split_string_list(write_raw_cpo_list[i],':');
+
+              AssertThrow(split_raw_cpo_instructions.size() == 2,
+                          ExcMessage("Value \""+ write_raw_cpo_list[i] +"\", set in \"Write out raw cpo data\", is not a correct option "
+                                     + "because it should contain a mineral identification and a output specifier seprated by a colon (:). This entry "
+                                     + "does not follow those rules."));
+
+              // get mineral number
+              std::vector<std::string> mineralnstructions = Utilities::split_string_list(split_raw_cpo_instructions[0],' ');
+              AssertThrow(mineralnstructions.size() == 2,
+                          ExcMessage("Value \""+ write_raw_cpo_list[i] +"\", set in \"Write out raw cpo data\", is not a correct option "
+                                     + "because the mineral identification part should contain two elements, the word mineral and a number."));
+
+              AssertThrow(mineralnstructions[0] == "mineral",
+                          ExcMessage("Value \""+ write_raw_cpo_list[i] +"\", set in \"Write out raw cpo data\", is not a correct option "
+                                     + "because the mineral identification part should start with the word mineral and it starts with \""
+                                     + mineralnstructions[0] + "\"."));
+
+              int mineral_number = Utilities::string_to_int(mineralnstructions[1]);
```

Comment:
why not `unsigned int`?

Reply:
There is no string_to_unsigned_int in dealii utilities. I could make a pull request to add it, but it would take a year until we can start using it.